assets/src/components/editing/elements/table/table-util.ts

```diff
@@ -0,0 +1,60 @@
+import * as ContentModel from 'data/content/model/elements/types';
+
+/**
+ * Given a set of model properties, returns a map of the appropriate
+ * attributes to set on a <td> or <th> element.
+ *
+ */
+export const cellAttributes = (attrs: ContentModel.TableHeader | ContentModel.TableData) => {
+  return {
+    className: attrs.align ? `text-${attrs.align}` : undefined,
+    colSpan: attrs.colspan || undefined,
+    rowSpan: attrs.rowspan || undefined,
+  };
+};
+
+export const getColspan = (cell: ContentModel.TableCell): number => cell.colspan || 1;
+export const getRowspan = (cell: ContentModel.TableCell): number => cell.rowspan || 1;
+
+/**
```

Comment:
I tried a few implementations of the whole merge/split trying to just use the slate location's, that turned out to be hard to get right and really hard to read. Came up with a solution to just calculate the visual grid and that made everything a lot easier.